simulators/optimism/daisy-chain/main.go

```diff
@@ -0,0 +1,266 @@
+package main
+
+import (
+	"context"
+	"math/big"
+	"math/rand"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/hive/hivesim"
+	"github.com/ethereum/hive/optimism"
+	"github.com/stretchr/testify/require"
+)
+
+// Test constants
+var (
+	// GoerliSequencerRPC is the RPC endpoint of the Goerli sequencer.
+	GoerliSequencerRPC = "https://goerli-sequencer.optimism.io"
+
+	// HistoricalSequencerRPC is the RPC endpoint of the historical sequencer on Goerli.
+	HistoricalSequencerRPC = "https://goerli-historical-0.optimism.io"
+
+	// MockAddr is a mock address used in the tests.
+	MockAddr = common.HexToAddress("0x00000000000000000000000000000000000badc0de")
+
+	// IDPrecompile is the address of the identity precompile. Used for gas estimation.
+	IDPrecompile = common.HexToAddress("0x0000000000000000000000000000000000000004")
+)
+
+var tests = []*optimism.TestSpec{
+	{Name: "daisy-chain-debug_traceBlockByNumber", Run: debugTraceBlockByNumberTest},
+	{Name: "daisy-chain-debug_traceBlockByHash", Run: debugTraceBlockByHashTest},
+	{Name: "daisy-chain-debug_traceTransaction", Run: debugTraceTransactionTest},
+	{Name: "daisy-chain-debug_traceCall", Run: debugTraceCallTest},
+	{Name: "daisy-chain-eth_call", Run: ethCallTest},
+	{Name: "daisy-chain-eth_estimateGas", Run: ethEstimateGasTest},
+}
+
+func main() {
+	sim := hivesim.New()
+	for _, forkName := range optimism.AllOptimismForkConfigs {
```

Comment:
I'm not sure we need to run this for each fork config given we're testing pre-bedrock content anyway. The active fork at the head of the chain won't matter.